app/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/demo/Example9Fragment.java

```diff
@@ -0,0 +1,203 @@
+                contacts.add(contact);
+            }
+        }
+
+        return contacts;
+    }
+
+
+    private class ContactsSection extends StatelessSection {
+
+        String title;
+        List<String> list;
+
+        ContactsSection(String title, List<String> list) {
+            super(new SectionParameters.Builder(R.layout.section_ex1_item)
+                    .headerResourceId(R.layout.section_ex1_header)
+                    .build());
+
+            this.title = title;
+            this.list = list;
+        }
+
+        @Override
+        public int getContentItemsTotal() {
+            return list.size();
+        }
+
+        @Override
+        public RecyclerView.ViewHolder getItemViewHolder(View view) {
+            return new ItemViewHolder1(view);
+        }
+
+        @Override
+        public void onBindItemViewHolder(RecyclerView.ViewHolder holder, int position) {
+            try {
+                if (custom1.isInstanceOf(holder)) {
+                    final ItemViewHolder1 itemHolder = (ItemViewHolder1) holder;
+
+                    String name = list.get(position);
+
+                    itemHolder.tvItem.setText(name);
+                    itemHolder.imgItem.setImageResource(name.hashCode() % 2 == 0 ? R.drawable.ic_face_black_48dp : R.drawable.ic_tag_faces_black_48dp);
+
+                    itemHolder.rootView.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            Toast.makeText(getContext(), String.format("Clicked on position #%s of Section %s, VH1", sectionAdapter.getPositionInSection(itemHolder.getAdapterPosition()), title), Toast.LENGTH_SHORT).show();
+                        }
+                    });
+                } else if (custom2.isInstanceOf(holder)) {
+                    final ItemViewHolder2 itemHolder = (ItemViewHolder2) holder;
+
+                    String name = list.get(position);
+
+                    itemHolder.tvItem.setText(name);
+                    itemHolder.imgItem.setImageResource(name.hashCode() % 2 == 0 ? R.drawable.ic_face_black_48dp : R.drawable.ic_tag_faces_black_48dp);
+
+                    itemHolder.rootView.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            Toast.makeText(getContext(), String.format("Clicked on position #%s of Section %s, VH2", sectionAdapter.getPositionInSection(itemHolder.getAdapterPosition()), title), Toast.LENGTH_SHORT).show();
+                        }
+                    });
+                } else {
+                    throw new Exception("Unrecognized ViewHolder");
+                }
+            } catch (Exception e) {
+                Log.e(this.getClass().getSimpleName(), "onBindItemViewHolder", e);
+            }
+        }
+
+        @Override
+        public int getItemViewType(int position) {
```

Comment:
The reason for the user handling this logic is because the ViewType to return for each item depends on the position of the item within the Section. For example, one may have a list of PhotoItems and TextItems within a single Section. Both PhotoItem and TextItem extend Item. The Section contains this list of Items, and when assigning ViewTypes, simply looks to see if the Item at position i is an instance of PhotoItem or TextItem and returns the applicable ViewType.